packages/web-client/tests/unit/services/hub-authentication-test.ts

```diff
@@ -0,0 +1,234 @@
+      .returns(Promise.resolve(false));
+    let authenticateStub = sinon
+      .stub(layer2Service, 'authenticate')
+      .returns(Promise.resolve(HUB_AUTH_TOKEN));
+    await hubAuthentication.ensureAuthenticated();
+
+    assert.ok(
+      authenticateStub.calledOnce,
+      'Called the authenticate method to get a new auth token'
+    );
+    assert.ok(
+      hubAuthentication.authToken === HUB_AUTH_TOKEN &&
+        hubAuthentication.isAuthenticated,
+      'authenticated'
+    );
+  });
+
+  test("it can fetch a new auth token when one doesn't exist", async function (assert) {
+    window.TEST__AUTH_TOKEN = undefined;
+    hubAuthentication = this.owner.lookup('service:hub-authentication');
+    await settled();
+
+    assert.ok(
+      !hubAuthentication.authToken && !hubAuthentication.isAuthenticated,
+      'not authenticated'
+    );
+
+    let authenticateStub = sinon
+      .stub(layer2Service, 'authenticate')
+      .returns(Promise.resolve(HUB_AUTH_TOKEN));
+    await hubAuthentication.ensureAuthenticated();
+
+    assert.ok(
+      authenticateStub.calledOnce,
+      'Called the authenticate method to get a new auth token'
+    );
+    assert.ok(
+      hubAuthentication.authToken === HUB_AUTH_TOKEN &&
+        hubAuthentication.isAuthenticated,
+      'authenticated'
+    );
+  });
+
+  test('it throws an error when fetching a new auth token fails with an empty string', async function (assert) {
+    hubAuthentication = this.owner.lookup('service:hub-authentication');
+    await settled();
+
+    assert.ok(
+      hubAuthentication.authToken === HUB_AUTH_TOKEN &&
+        hubAuthentication.isAuthenticated,
+      'authenticated'
+    );
+
+    sinon.stub(layer2Service, 'authenticate').returns(Promise.resolve(''));
+    sinon
+      .stub(layer2Service, 'checkHubAuthenticationValid')
+      .returns(Promise.resolve(false));
+    let error = null;
+    try {
+      await hubAuthentication.ensureAuthenticated();
+    } catch (e) {
+      error = e;
+    }
+
+    assert.equal(
+      error.message,
+      'Failed to fetch auth token',
+      'It fails with the error message for a falsey auth token'
+    );
+    assert.ok(
+      !hubAuthentication.authToken && !hubAuthentication.isAuthenticated,
+      'not authenticated'
+    );
+  });
+
+  test('it throws an error when errors are thrown while fetching a new auth token', async function (assert) {
+    hubAuthentication = this.owner.lookup('service:hub-authentication');
+    await settled();
+
+    assert.ok(
+      hubAuthentication.authToken === HUB_AUTH_TOKEN &&
+        hubAuthentication.isAuthenticated,
+      'authenticated'
+    );
+
+    sinon.stub(layer2Service, 'authenticate').throws(new Error('A test error'));
+    sinon
+      .stub(layer2Service, 'checkHubAuthenticationValid')
+      .returns(Promise.resolve(false));
+    let error = null;
+    try {
+      await hubAuthentication.ensureAuthenticated();
+    } catch (e) {
+      error = e;
+    }
```

Comment:
I found this, but it didn't support async stuff. Reviewing the docs + github comments because of your comment made me find this though:

https://api.qunitjs.com/assert/rejects/

Reply:
Will update to use rejects